services/src/main/scala/cromwell/services/healthmonitor/impl/workbench/WorkbenchHealthMonitorServiceActor.scala

```diff
@@ -0,0 +1,87 @@
+package cromwell.services.healthmonitor.impl.workbench
+
+import java.net.URL
+
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.config.Config
+import cromwell.cloudSupport.gcp.GoogleConfiguration
+import cromwell.cloudSupport.gcp.gcs.GcsStorage
+import cromwell.cloudSupport.gcp.genomics.GenomicsFactory
+import cromwell.core.WorkflowOptions
+import cromwell.services.healthmonitor.HealthMonitorServiceActor
+import cromwell.services.healthmonitor.HealthMonitorServiceActor.{MonitoredSubsystem, OkStatus, SubsystemStatus}
+import net.ceedubs.ficus.Ficus._
+import cats.instances.future._
+import cats.syntax.functor._
+import cromwell.services.healthmonitor.impl.common.{DockerHubMonitor, EngineDatabaseMonitor}
+
+import scala.concurrent.Future
+
+/**
+  * A health monitor implementation which will monitor external services used in the Workbench ecosystem, such
+  * as GCS and PAPI. This implementation makes some assumptions of Cromwell's configuration which will be true
+  * in a Workbench scenario but YMMV otherwise. Caveat emptor and all of that fun stuff.
+  */
+class WorkbenchHealthMonitorServiceActor(val serviceConfig: Config, globalConfig: Config)
+  extends HealthMonitorServiceActor
+    with DockerHubMonitor
+    with EngineDatabaseMonitor {
+  override implicit val system = context.system
+
+  override lazy val subsystems: Set[MonitoredSubsystem] = Set(DockerHub, EngineDb, Papi, Gcs)
+
+  private lazy val Gcs = MonitoredSubsystem("GCS", checkGcs _)
+  private lazy val Papi = MonitoredSubsystem("PAPI", checkPapi _)
+
+  val googleConfig = GoogleConfiguration(globalConfig)
+  val papiBackendName = serviceConfig.as[Option[String]]("services.HealthMonitor.config.PapiBackendName").getOrElse("Jes")
+  val papiConfig: Config = globalConfig.as[Config]("backend.providers.Jes.config")
+
+  /**
+    * Demonstrates connectivity to GCS by stat-ing a public bucket. We use a public bucket as our plethora of auth
+    * modes make it impossible to ensure we would always be able to ping an arbitrary bucket with the credentials
+    * available.
+    */
+  private def checkGcs(): Future[SubsystemStatus] = {
+    import WorkbenchHealthMonitorServiceActor.GcsBucketToCheck
+
+    // For any expected production usage of this check, the GCS bucket should be public read */
+    val gcsBucketToCheck = serviceConfig.as[String]("services.HealthMonitor.config.GcsBucketToCheck")
+
+    val gcsAuthName = papiConfig.as[String]("filesystems.gcs.auth")
+
+    val cred = googleConfig.auth(gcsAuthName) match {
+      case Valid(a) => a.credential(WorkflowOptions.empty)
```

Comment:
Sorry to be so annoying with this -_- But in the current FC config (granted it's going to change soon when we get rid of refresh tokens), the `filesystems.gcs.auth` is going to be a "refresh token" auth mode, which will need a refresh token that it expects to get from workflow options, so I don't *think* this will work (I'm happy to give it a shot with a refresh token config if you want).
Basically since there is only one of those actors it cannot depend on a per workflow or per user type of auth.

I would suggest to just use application-default here since the bucket is expected to be public, or have a separate config stanza that defines what the auth for this check should be.
It will also have the added benefit to keep this GCS check working even without a JES backend configured.

Reply:
You mean have this implementation require the existence of an auth other than refresh token (e.g. app default)? If I understood you that seems like a good idea to me

Reply:
yes exactly